### DISCUSSION TITLE
Combine Different Images to One, part 3

### DISCUSSION THREAD

**Discussion by Jay123210599** [ImageMagick/ImageMagick#7363] [Help]
### ImageMagick version

7.1.1-27

### Operating system, version and so on

Windows 11

### Description

I am combining images to do a manga and anime comparison for a series. However, the final comparison image had a white space at the bottom. I also want to make sure each individual image to goes with the right counterpart without them overlapping (e.g. manga1.png to anime1png, manga2.png to anime2.png, etc). How do I do that?

This is what I used for the final result: `magick manga.png anime.png -resize "%[fx:max(u.w,v.w)]x" +append comparison.png`

![comparison](https://github.com/ImageMagick/ImageMagick/assets/141977822/b3ce46ab-8296-497e-9f5c-6d534c4bbbfc)
[manga and anime.zip](https://github.com/user-attachments/files/15526482/manga.and.anime.zip)

**Comment by snibgo** [comment]
Your command matches the *width* of the two inputs. So their *heights* are different. When placed side-by-side, the heights are different, and IM fills the gap with the background colour, which is white.

Instead, you could match the *heights*:
```
  magick manga.png anime.png -resize "x%[fx:max(u.h,v.h)]" +append c2.png
```

For processing many pairs of inputs, you could have a shell script, such as this (Windows BAT):
```
setlocal enabledelayedexpansion

for %%F in (manga*.png) do (
  echo %%F
  set MANG=%%F
  set ANIM=!MANG:manga=anime!
  set OUT=!MANG:manga=out!
  echo !MANG! !ANIM! !OUT!

  magick !MANG! !ANIM! -resize "x%%[fx:max(u.h,v.h)]" +append !OUT!

)
```
